### ISSUE TITLE
isGooglePaySupported causes crash in android

### ISSUE THREAD

**Issue by kostiantyn-solianyk** [stripe/stripe-react-native#1065]
**Describe the bug**
App is wrapped in the `StripeProvider` and in the child component I'm calling `isGooglePaySupported` promise, but each time it causes crash in android.
`java.lang.IllegalStateException: PaymentConfiguration was not initialized. Call PaymentConfiguration.init().`

In manifest I have
```
<meta-data
      android:name="com.google.android.gms.wallet.api.enabled"
      android:value="true" />
```

**Smartphone (please complete the following information):**
 - Device: android (real and emulator)
 - OS: 10
 - Stripe: 0.15.0
 - RN: 0.68.2
 
 Case that I have:
```
 <StripeProvider publishableKey={stripePublishableKey} merchantIdentifier={merchantIdentifier}>
     <ChildComponent />
 </StripeProvider>
```
 
 Inside `ChildComponent` I'm calling 
```
useEffect(() => {
  async function igGPaySupported() {
    const isSupported = await isGooglePaySupported({ testEnv: true });
  }
  igGPaySupported();
}, []);
```

 and `isGooglePaySupported` causes a crash

**Comment by charliecruzan-stripe** [comment]
We call `PaymentConfiguration.init` behind the scenes in the initialize function (called by the StripeProvider), so it looks like there's a race condition where that call hasn't completed yet and you're calling out to `isGooglePaySupported`. 

Could you confirm that for me?

**Comment by kartavyaparekh96** [comment]
+1

**Comment by Rc85** [comment]
Having the same issue only on first build. After the crash, the issue goes away when reopening app. I assume init is called at that point.
```
<StripeProvider
publishableKey={stripeApiKey}
merchantIdentifier='package.com'
urlScheme='mytakeouts-payment'
setReturnUrlSchemeOnAndroid
>
  <Payment /> // isGoogleSupported sits in here
</StripeProvider>
```
The way React mounts components is always children first, so this is probably the issue for people who have `isGoogleSupported` in a child component of `StripeProvider`. Looks like @kostiantyn-solianyk suggested `setTimeout` is the current workaround.

**Comment by b13i** [comment]
@charliecruzan-stripe any updates from the team on getting this on the roadmap to fix? Glad there's a short-term workaround but would love to proactively address this in the library if possible

**Comment by charliecruzan-stripe** [comment]
Yep, this will get fixed soon. Have a branch open to fix this and other cases soon

**Comment by NathHorrigan** [comment]
@charliecruzan-stripe  - Any update here? This is breaking our app in production.

**Comment by sregg** [comment]
Any update on this? Any workaround possible while waiting for a proper fix?
We don't use `StripeProvider` and call `initStripe` only further in the app once we know the user's country (i.e. after calling `isPlatformPaySupported`).

**Comment by NathHorrigan** [comment]
Hello @sregg ,

I had the exact same problem, don't init until we know the user's country which requires waiting until the store was loaded from storage.

I found the best solution was to wrap the `initStripe` call in a function that accepts an option country param, if the param is null then assume US/UK/FR etc, you can then call this function in your index.js file before any of your app code executes.

Once your app is loaded you can recall your initStripe method with the correct country, It's not a perfect approach but we are using this in production just fine.

**Comment by sregg** [comment]
Thanks Nathan. That's pretty much what I ended up doing.

**Comment by joe-yeomans** [comment]
@charliecruzan-stripe Is there any update on this? We are experiencing this issue as well. Would be good to know where you are with the issue. 
Thanks!

**Comment by trilam1409** [comment]
I got an issue because I called `isPlatformPaySupported` before there was a publishable key. Please double-check if you got the issue like me.

**Comment by AndrejStarusev** [comment]
> I got an issue because I called `isPlatformPaySupported` before there was a publishable key. Please double-check if you got the issue like me.

Many thanks, it was my issue as well

**Comment by FavazSlider** [comment]
> I got an issue because I called `isPlatformPaySupported` before there was a publishable key. Please double-check if you got the issue like me.

great bro, it worked for me,what i did is called isPlatformPaySupported in the then block of stripe init like below

      initStripe({
        publishableKey: appData?.profile?.preferences?.stripe_publishable_key,
        merchantIdentifier: "merchant.identifier",
      }).then(async () => {
        await isPlatformPaySupported().then((res) => {
          setIsApplePaySupported(red);
        });
      });
